src/coreclr/src/jit/optimizer.cpp

```diff
@@ -8659,6 +8659,163 @@ bool Compiler::optIsRangeCheckRemovable(GenTree* tree)
     return true;
 }
 
+//----------------------------------------------------------------------------------
+// optBranchlessConditions: optimize simple expressions to branchless expressions
+//
+//  Operation:
+//      Look for BBJ_COND basic blocks which target two BBJ_RETURN blocks
+//      and convert them into single branchless expressions, e.g.
+//
+//            JTRUE
+//              \
+//              EQ              RETURN
+//             /  \                \
+//            X    Y               ADD
+//                       =>       /   \
+//           RETURN             EQ     C3
+//              \              /  \
+//              C1            X    Y
+//
+//           RETURN
+//              \
+//               C2
+//
+//  Note:
+//      At the moment, the optimization works only for simple BBJ_RETURN blocks
+//      and returned values are of difference of 1.
+//
+void Compiler::optBranchlessConditions()
+{
+    JITDUMP("\n*************** In optBranchlessConditions()\n");
+    bool changed = false;
+    for (BasicBlock* block = fgFirstBB; block; block = block->bbNext)
+    {
+        if ((block->bbJumpKind != BBJ_COND))
+        {
+            continue;
+        }
+
+        Statement* firstStmt = block->lastStmt();
```

Comment:
why the last statement is called first?

Reply:
This is how I skip possible leading `GT_ASG` statements